<h1>Activity 2 - UK ROAD ACCIDENT DATA ANALYTICS PROJECT</h1>
<hr>
<h1>Analyst : Joshua Z. Deñado</h1>

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy.stats import f_oneway
import warnings
warnings.filterwarnings('ignore')

In [2]:
uk_road_accident = pd.read_csv('datasets\\uk_road_accident.csv')
uk_road_accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [3]:
uk_road_accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [4]:
uk_road_accident.describe().T

count       mean       std        min        25%  \
Latitude              660654.0  52.553866  1.406922  49.914430  51.490690   
Longitude             660653.0  -1.431210  1.383330  -7.516225  -2.332291   
Number_of_Casualties  660679.0   1.357040  0.824847   1.000000   1.000000   
Number_of_Vehicles    660679.0   1.831255  0.715269   1.000000   1.000000   

                            50%        75%        max  
Latitude              52.315641  53.453452  60.757544  
Longitude             -1.411667  -0.232869   1.762010  
Number_of_Casualties   1.000000   1.000000  68.000000  
Number_of_Vehicles     2.000000   2.000000  32.000000

<h2>Checking Null Values</h2>

In [5]:
uk_road_accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

<h2>Cleaning the Data Frame</h2>

In [6]:
uk_road_accident['Latitude'] = uk_road_accident['Latitude'].fillna(uk_road_accident['Latitude'].mean())
uk_road_accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [7]:
uk_road_accident['Longitude'] = uk_road_accident['Longitude'].fillna(uk_road_accident['Longitude'].mean())
uk_road_accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                      0
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [8]:
uk_road_accident['Road_Surface_Conditions'].mode()

0    Dry
Name: Road_Surface_Conditions, dtype: object

In [9]:
uk_road_accident['Road_Surface_Conditions'] = uk_road_accident['Road_Surface_Conditions'].fillna(uk_road_accident['Road_Surface_Conditions'].mode()[0])
uk_road_accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                      0
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions        0
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [10]:
uk_road_accident['Road_Type'].mode()

0    Single carriageway
Name: Road_Type, dtype: object

In [11]:
uk_road_accident['Road_Type'] = uk_road_accident['Road_Type'].fillna(uk_road_accident['Road_Type'].mode()[0])
uk_road_accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                      0
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions        0
Road_Type                      0
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [12]:
uk_road_accident['Urban_or_Rural_Area'].mode()

0    Urban
Name: Urban_or_Rural_Area, dtype: object

In [13]:
uk_road_accident['Urban_or_Rural_Area'] = uk_road_accident['Urban_or_Rural_Area'].fillna(uk_road_accident['Urban_or_Rural_Area'].mode()[0])
uk_road_accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                      0
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions        0
Road_Type                      0
Urban_or_Rural_Area            0
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [14]:
uk_road_accident['Weather_Conditions'].mode()

0    Fine no high winds
Name: Weather_Conditions, dtype: object

In [15]:
uk_road_accident['Weather_Conditions'] = uk_road_accident['Weather_Conditions'].fillna(uk_road_accident['Weather_Conditions'].mode()[0])
uk_road_accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

<h2>Checking the Data Types</h2>

In [16]:
uk_road_accident.dtypes

Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

In [17]:
uk_road_accident['Urban_or_Rural_Area'] = uk_road_accident['Urban_or_Rural_Area'].astype('object')
uk_road_accident.dtypes

Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

<h2>Changing the Data Types</h2>

In [18]:
uk_road_accident['Index'] = uk_road_accident['Index'].astype('category')
uk_road_accident['Accident_Severity'] = uk_road_accident['Accident_Severity'].astype('category')
uk_road_accident['Accident Date'] = uk_road_accident['Accident Date'].astype('category')
uk_road_accident['Light_Conditions'] = uk_road_accident['Light_Conditions'].astype('category')
uk_road_accident['District Area'] = uk_road_accident['District Area'].astype('category')
uk_road_accident['Road_Surface_Conditions'] = uk_road_accident['Road_Surface_Conditions'].astype('category')
uk_road_accident['Road_Type'] = uk_road_accident['Road_Type'].astype('category')
uk_road_accident['Urban_or_Rural_Area'] = uk_road_accident['Urban_or_Rural_Area'].astype('category')
uk_road_accident['Weather_Conditions'] = uk_road_accident['Weather_Conditions'].astype('category')
uk_road_accident['Vehicle_Type'] = uk_road_accident['Vehicle_Type'].astype('category')
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

<h1 style="color:red;">Explanatory Data Analytics (EDA)</h1>

Make 20 Questions

1. Which type of vehicle encounter most accidents?
2. Which type of weather condtions does accident mostly occur?
3. Does accidents commonly happen in Urban areas or in Rural areas?
4. Which type of road does accidents commonly happen?
5. Which road surface conditions does accident mostly happen?
6. which type of vehicle does have the least recorded serious accident severity?
7. Which light conditions contributes the most serious accidents?
8. Which district area does most accidents happen?
9. How many serious accidents happen in Birmingham District
10. What type of vehicle contribute the most in accidents in Birmingham?
11. What weather conditions does accidents happen in Clackmannanshire?
12. How many fatal accidents happen in Clackmannanshire District?
13. What weather conditions does accidents happen in Clackmannanshire?
14. What type of weather conditions does least amount car accidents happen?
15. How many casualties happen in Bermingham?
16. Which type of road surface conditions does accidents happen in Birmingham?
17. Which type of road surface conditions does accidents happen in Clackmannanshire?
18. Which type of road surface conditions does accidents most happen in Birmingham District?
19. Which type of road does accidents most happen Cars?
20. Which type of road surface conditions does accidents most happen Cars?

<h2>1. Which type of vehicle is involved in the highest number of accidents?</h2>

In [19]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [20]:
vehicle = uk_road_accident['Vehicle_Type'].value_counts()
vehicle

Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64

INSIGHTS : Cars are the most frequently involved vehicles, with 497,992 recorded accidents.

<h2>2. Under what weather conditions do accidents occur most frequently?</h2>

In [21]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [22]:
uk_road_accident['Weather_Conditions'].unique()

['Fine no high winds', 'Raining no high winds', 'Other', 'Fine + high winds', 'Raining + high winds', 'Snowing no high winds', 'Fog or mist', 'Snowing + high winds']
Categories (8, object): ['Fine + high winds', 'Fine no high winds', 'Fog or mist', 'Other', 'Raining + high winds', 'Raining no high winds', 'Snowing + high winds', 'Snowing no high winds']

In [23]:
weather = uk_road_accident['Weather_Conditions'].value_counts()
weather

Weather_Conditions
Fine no high winds       535013
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

INSIGHTS : The majority of accidents happen during fine weather with no strong winds (535,013 accidents). This indicates that most accidents are not caused by adverse weather, but by other factors such as driver behavior.

<h2>3. Are accidents more common in urban areas or rural areas?</h2>

In [24]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [25]:
uk_road_accident['Urban_or_Rural_Area'].unique()

['Urban', 'Rural', 'Unallocated']
Categories (3, object): ['Rural', 'Unallocated', 'Urban']

In [26]:
accidents = uk_road_accident['Urban_or_Rural_Area'].value_counts()
accidents

Urban_or_Rural_Area
Urban          421678
Rural          238990
Unallocated        11
Name: count, dtype: int64

INSIGHTS : Accidents are more common in urban areas (421,678 accidents). This may be due to higher traffic density and more complex road systems in cities.

<h2>4. On which type of road do accidents most often occur?</h2>

In [27]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [28]:
uk_road_accident['Road_Type'].unique()

['Single carriageway', 'Dual carriageway', 'One way street', 'Roundabout', 'Slip road']
Categories (5, object): ['Dual carriageway', 'One way street', 'Roundabout', 'Single carriageway', 'Slip road']

In [29]:
road_type = uk_road_accident['Road_Type'].value_counts()
road_type

Road_Type
Single carriageway    496663
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64

INSIGHTS : Single carriageways record the highest number of accidents, with 496,663 cases. This could be linked to their heavy use and limited separation of traffic directions.

<h2>5. Under what road surface conditions do most accidents occur?</h2>

In [30]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [31]:
uk_road_accident['Road_Surface_Conditions'].unique()

['Dry', 'Wet or damp', 'Snow', 'Frost or ice', 'Flood over 3cm. deep']
Categories (5, object): ['Dry', 'Flood over 3cm. deep', 'Frost or ice', 'Snow', 'Wet or damp']

In [32]:
road_surface = uk_road_accident['Road_Surface_Conditions'].value_counts()
road_surface

Road_Surface_Conditions
Dry                     448547
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

INSIGHTS : Most accidents happen on dry roads (448,547 accidents). This shows that hazardous conditions like rain or ice are not the main contributors to accident frequency.

<h2>6. Which type of vehicle records the fewest serious accidents?</h2>

In [33]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [34]:
uk_road_accident['Accident_Severity'].unique()

['Serious', 'Slight', 'Fatal']
Categories (3, object): ['Fatal', 'Serious', 'Slight']

In [35]:
serious_road_accidents = uk_road_accident[uk_road_accident['Accident_Severity'] == 'Serious'] #just fileters out the Adelie species

types_vehicle = serious_road_accidents.groupby(['Vehicle_Type']).size()
types_vehicle

Vehicle_Type
Agricultural vehicle                       282
Bus or coach (17 or more pass seats)      3373
Car                                      66461
Data missing or out of range                 0
Goods 7.5 tonnes mgw and over             2321
Goods over 3.5t. and under 7.5t            857
Minibus (8 - 16 passenger seats)           276
Motorcycle 125cc and under                2031
Motorcycle 50cc and under                 1014
Motorcycle over 125cc and up to 500cc     1014
Motorcycle over 500cc                     3457
Other vehicle                              767
Pedal cycle                                 39
Ridden horse                                 0
Taxi/Private hire car                     1771
Van / Goods 3.5 tonnes mgw or under       4554
dtype: int64

INSIGHTS : Pedal cycles have the fewest serious accidents, with only 39 recorded cases.

<h2>7. What lighting conditions are linked to the fewest serious accidents?</h2>

In [36]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [37]:
uk_road_accident['Accident_Severity'].unique()

['Serious', 'Slight', 'Fatal']
Categories (3, object): ['Fatal', 'Serious', 'Slight']

In [38]:
serious_road_accidents = uk_road_accident[uk_road_accident['Accident_Severity'] == 'Serious'] #just fileters out the Adelie species

light_conditions = serious_road_accidents.groupby(['Light_Conditions']).size()
light_conditions

Light_Conditions
Darkness - lighting unknown      794
Darkness - lights lit          19130
Darkness - lights unlit          360
Darkness - no lighting          7174
Daylight                       60759
dtype: int64

INSIGHTS : The lowest number of serious accidents occur in Darkness - lights unlit roads (360 cases).

<h2>8. In which district do the highest number of accidents take place?</h2>

In [39]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [40]:
district_area = uk_road_accident['District Area'].value_counts()
district_area

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

INSIGHTS : The district of Birmingham records the most accidents, with 13,491 incidents.

<h2>9. How many serious accidents occur in the Birmingham district?</h2>

In [41]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [42]:
serious_road_accidents = uk_road_accident[uk_road_accident['District Area'] == 'Birmingham'] #just fileters out the Adelie species

severity = serious_road_accidents.groupby(['Accident_Severity']).size()
severity

Accident_Severity
Fatal        105
Serious     1474
Slight     11912
dtype: int64

INSIGHTS : Birmingham has 1,474 serious accidents on record.

<h2>10. Which type of vehicle is least often involved in accidents in Birmingham??</h2>

In [43]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [44]:
road_district = uk_road_accident[uk_road_accident['District Area'] == 'Birmingham'] #just fileters out the Adelie species

vehicle_type = road_district.groupby(['Vehicle_Type']).size()
vehicle_type

Vehicle_Type
Agricultural vehicle                       61
Bus or coach (17 or more pass seats)      570
Car                                      9600
Data missing or out of range                0
Goods 7.5 tonnes mgw and over             463
Goods over 3.5t. and under 7.5t           132
Minibus (8 - 16 passenger seats)           43
Motorcycle 125cc and under                362
Motorcycle 50cc and under                 134
Motorcycle over 125cc and up to 500cc     133
Motorcycle over 500cc                     578
Other vehicle                             113
Pedal cycle                                13
Ridden horse                                0
Taxi/Private hire car                     504
Van / Goods 3.5 tonnes mgw or under       785
dtype: int64

INSIGHTS : Pedal cycles are the least involved, with just 13 accidents. This may be due to fewer cyclists compared to cars and other vehicles.

<h2>11. What road surface conditions are most often linked to accidents in Birmingham?</h2>

In [45]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [46]:
road_district = uk_road_accident[uk_road_accident['District Area'] == 'Birmingham'] #just fileters out the Adelie species

road_condition = road_district.groupby(['Road_Surface_Conditions']).size()
road_condition

Road_Surface_Conditions
Dry                     9367
Flood over 3cm. deep       8
Frost or ice             306
Snow                     109
Wet or damp             3701
dtype: int64

INSIGHTS : Most accidents in Birmingham happen on dry roads (9,367 accidents).

<h2>12. Under which weather conditions do most accidents occur in Birmingham?</h2>

In [47]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [48]:
road_district = uk_road_accident[uk_road_accident['District Area'] == 'Birmingham'] #just fileters out the Adelie species

weather_condition = road_district.groupby(['Weather_Conditions']).size()
weather_condition

Weather_Conditions
Fine + high winds          125
Fine no high winds       10638
Fog or mist                 37
Other                      713
Raining + high winds       126
Raining no high winds     1717
Snowing + high winds         8
Snowing no high winds      127
dtype: int64

INSIGHTS : Fine weather with no strong winds accounts for 10,638 accidents in Birmingham. This again shows that accidents happen mainly in normal driving conditions rather than extreme weather.

<h2>13. How many fatal accidents are recorded in the Clackmannanshire district?</h2>

In [49]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [50]:
road_district = uk_road_accident[uk_road_accident['District Area'] == 'Clackmannanshire'] #just fileters out the Adelie species

accident_severity = road_district.groupby(['Accident_Severity']).size()
accident_severity

Accident_Severity
Fatal       1
Serious    18
Slight     72
dtype: int64

INSIGHTS : Only 1 fatal accident is reported in Clackmannanshire. This suggests that while accidents occur, very few result in fatalities.

<h2>14. Which road surface condition is least associated with accidents in Clackmannanshire?</h2>

In [51]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [52]:
road_district = uk_road_accident[uk_road_accident['District Area'] == 'Clackmannanshire'] #just fileters out the Adelie species

road_condition = road_district.groupby(['Road_Surface_Conditions']).size()
road_condition

Road_Surface_Conditions
Dry                     56
Flood over 3cm. deep     0
Frost or ice             4
Snow                     0
Wet or damp             31
dtype: int64

INSIGHTS : Frost or ice is linked to just 4 accidents.

<h2>15. Which road surface condition is associated with the very fewest accidents in Clackmannanshire?</h2>

In [53]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [54]:
road_district = uk_road_accident[uk_road_accident['District Area'] == 'Clackmannanshire'] #just fileters out the Adelie species

weather_condition = road_district.groupby(['Weather_Conditions']).size()
weather_condition

Weather_Conditions
Fine + high winds         3
Fine no high winds       71
Fog or mist               1
Other                     5
Raining + high winds      0
Raining no high winds    11
Snowing + high winds      0
Snowing no high winds     0
dtype: int64

INSIGHTS : Fog or mist has the lowest number, with only 1 accident recorded.

<h2>16. In what type of weather do the fewest car accidents occur?</h2>

In [55]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [56]:
road_district = uk_road_accident[uk_road_accident['Vehicle_Type'] == 'Car'] #just fileters out the Adelie species

weather_condition = road_district.groupby(['Weather_Conditions']).size()
weather_condition

Weather_Conditions
Fine + high winds          6463
Fine no high winds       403324
Fog or mist                2641
Other                     12993
Raining + high winds       7206
Raining no high winds     59940
Snowing + high winds        677
Snowing no high winds      4748
dtype: int64

INSIGHTS : The lowest number of car accidents happen in snow with strong winds (677 accidents). Extreme weather may discourage travel, lowering accident rates.

<h2>17. In what type of weather do fewest bus accidents occur?</h2>

In [57]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [58]:
vehicle_type = uk_road_accident[uk_road_accident['Vehicle_Type'] == 'Bus or coach (17 or more pass seats)'] #just fileters out the Adelie species

weather_condition = vehicle_type.groupby(['Weather_Conditions']).size()
weather_condition

Weather_Conditions
Fine + high winds          350
Fine no high winds       20963
Fog or mist                134
Other                      609
Raining + high winds       389
Raining no high winds     3182
Snowing + high winds        32
Snowing no high winds      219
dtype: int64

INSIGHTS : Snowing + high winds.

<h2>18. How many 1 casualties are recorded in Birmingham?</h2>

In [59]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [60]:
road_district = uk_road_accident[uk_road_accident['District Area'] == 'Birmingham'] #just fileters out the Adelie species

no_casualties = road_district.groupby(['Number_of_Casualties']).size()

print(no_casualties, no_casualties.sum())

Number_of_Casualties
1     10138
2      2238
3       681
4       276
5        88
6        47
7        14
8         4
9         2
10        1
15        1
18        1
dtype: int64 13491


INSIGHTS : Birmingham records 10,138 casualties in road accidents.

<h2>19. Birmingham records 10,138 casualties in road accidents?</h2>

In [61]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [62]:
road_district = uk_road_accident[uk_road_accident['District Area'] == 'Birmingham'] #just fileters out the Adelie species

road_type = road_district.groupby(['Road_Type']).size()
road_type

Road_Type
Dual carriageway      3039
One way street         321
Roundabout            1008
Single carriageway    9025
Slip road               98
dtype: int64

INSIGHTS : Single carriageways are the most accident-prone, with 9,025 incidents.

<h2>20. Which type of road accounts for the fewest accidents in Clackmannanshire?</h2>

In [63]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [64]:
road_district = uk_road_accident[uk_road_accident['District Area'] == 'Clackmannanshire'] #just fileters out the Adelie species

road_type = road_district.groupby(['Road_Type']).size()
road_type

Road_Type
Dual carriageway       7
One way street         2
Roundabout            12
Single carriageway    70
Slip road              0
dtype: int64

INSIGHTS : One-way streets have the lowest accident count, with only 2 cases.

<h2>21. On which type of road do car accidents occur least frequently?</h2>

In [65]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [66]:
vehicle_type = uk_road_accident[uk_road_accident['Vehicle_Type'] == 'Car'] #just fileters out the Adelie species

road_type = vehicle_type.groupby(['Road_Type']).size()
road_type

Road_Type
Dual carriageway       74820
One way street         10223
Roundabout             33187
Single carriageway    374464
Slip road               5298
dtype: int64

INSIGHTS : Car accidents are least common on slip roads (5,298 accidents).

<h2>22. Under which road surface conditions do car accidents occur least often?</h2>

In [67]:
uk_road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [68]:
vehicle_type = uk_road_accident[uk_road_accident['Vehicle_Type'] == 'Car'] #just fileters out the Adelie species

road_condition = vehicle_type.groupby(['Road_Surface_Conditions']).size()
road_condition

Road_Surface_Conditions
Dry                     337860
Flood over 3cm. deep       777
Frost or ice             14108
Snow                      4483
Wet or damp             140764
dtype: int64

INSIGHTS : The fewest car accidents occur on flooded roads over 3cm deep (777 cases).